keras_nlp/samplers/beam_sampler_test.py

```diff
@@ -67,6 +68,23 @@ def test_stateful_call(self):
         )
         self.assertEqual(self.join_as_string(output), ["sequentially"])
 
+    def test_return_all_beams(self):
+        state_chars = list("sequentially")
+        state = tf.constant([[self.char_lookup[c] for c in state_chars]])
+        prompt = tf.fill((self.batch_size, self.length), self.char_lookup["z"])
+        output = self.sampler_all_beams(
+            next=self.next,
+            prompt=prompt,
+            state=state,
+        )
+
+        self.assertEqual(output[0].shape, (self.batch_size, 5, self.length))
+        self.assertEqual(output[1].shape, (self.batch_size, 5))
+        self.assertTrue(tf.reduce_all(output[1][:, 1:] <= output[1][:, :-1]))
+        self.assertEqual(
```

Comment:
Let's also test `self.join_as_string(output[0][:, 0, :]) == ["sequentially"]` since we are testing returning all beams.